bitnami/redis-cluster/templates/update-cluster.yaml

```diff
@@ -108,12 +126,25 @@ spec:
                   echo "Node $new_node_ip not ready, waiting for all the nodes to be ready..."
                   sleep 5
                 done
-                firstNodeIp=$(wait_for_dns_lookup {{ template "common.names.fullname" . }}-0.{{ template "common.names.fullname" . }}-headless 120 5)
+                slave=()
+                if (( new_node_index % 2 )); then
+                  slave+=("--cluster-slave")
+                  [ -z previousNodeIp ] && previousNodeIp=$firstNodeIp
+                  masterID=$(redis-cli -c -h $previousNodeIp -a $REDISCLI_AUTH cluster myid)
+                  slave+=("--cluster-master-id $masterID")
+                  echo "Adding Node $new_node_index ${new_node_ip} as slave of ${masterID}"
+                else
+                  previousNodeIp=$new_node_ip
+                  echo "Adding Node $new_node_index ${new_node_ip} as master"
+                fi
```

Comment:
So looking at this logic (and similar code above), will this try to add slaves to a "master only" cluster (as described in #3853? We use redis cluster for a cache where if one of the masters goes down we basically go to the service for the data until the master is back up vs. have followers that can take their place.

Reply:
I will check that use case.
Do you mind sharing the parameters you use to launch the chart ?

Reply:
So I think if you do something like this:
1. `helm install redis-cluster bitnami/redis-cluster --set 'cluster.replicas=0'`
2. `helm upgrade redis-cluster bitnami/redis-cluster --set 'cluster.nodes=6,cluster.replicas=0,cluster.init=false,cluster.update.addNodes=true,cluster.update.currentNumberOfNodes=3'`
then it will try to make every other node a slave.

But I already see that you addressed this with https://github.com/bitnami/charts/pull/3928/commits/2e93a33f5a139651d817a653109bb5a0f3f4f8cb